constraint/pkg/client/template_client.go

```diff
@@ -60,6 +63,44 @@ func (e *templateClient) ValidateConstraint(constraint *unstructured.Unstructure
 	return crds.ValidateCR(constraint, e.crd)
 }
 
+// ApplyDefaultParams will apply any default parameters defined in the CRD of the constraint's
+// corresponding template.
+// Assumes ValidateConstraint() called is called so the constraint is a valid CRD.
+func (e *templateClient) ApplyDefaultParams(constraint *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	cpy := constraint.DeepCopy()
```

Comment:
We shouldn't deepcopy here, just modify the input value in-place. Let callers decide if deepcopy is necessary or if they'd prefer to avoid the performance penalty.